src/system/sys_profiler.cc

```diff
@@ -703,14 +701,20 @@ void Profiler::output(MPI_Comm comm, ostream &os) {
 }
 
 
-void Profiler::output(MPI_Comm comm) {
-    int mpi_rank;
-    chkerr( MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank) );
-    if (mpi_rank == 0) {
-        output(comm, *get_default_output_stream());
-    } else {
-        ostringstream os;
-        output(comm, os );
+void Profiler::output(MPI_Comm comm, string profiler_path /* = "" */) {
+    if(profiler_path == "") {
+        output(comm);
```

Comment:
`if (mpi_rank == 0)` should come first.
then use `*get_default_output_stream()` in the case of empty path.

Reply:
Done